### ISSUE TITLE
Search problems or request to customize search settings

### ISSUE THREAD

**Issue by amix307** [go-gitea/gitea#30064]
### Description

My repositories contain code for CI/CD system builds such as Jenkins or Teamcity.
Teamcity stores its code in the .teamcity directory, which can contain directories of several nesting levels, as well as xml, kotlin files.
The problem is that Gitea's built-in Bleeve indexer ignores this directory and I get no search results in it. On the other hand, if I store the parameters for Maven in the .mvn directory, the search is successful.

I suspect this is something that can be controlled or is a bug that I would like to fix. 

### Gitea Version

1.21.7

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Centos9 Stream

### How are you running Gitea?

on-prem

### Database

PostgreSQL

**Comment by lunny** [comment]
Can you give some example repositories?

**Comment by amix307** [comment]
@lunny Of course, but on try.gitea.com search func is off. In what form can I give the repo example?

**Comment by amix307** [comment]
@lunny for example, clone random repo with .teamcity directory from GH: https://github.com/jonnyzzz/TeamCity.Node
Then push it to your Gitea instance with enabled search.
Search something from .teamcity, for example "uploadedKey" from.teamcity/TeamCityNodePlugin/vcsRoots/git___github_com_jonnyzzz_TeamCity_Node_git.kt

**Comment by amix307** [comment]
@lunny Hi! Will you be able to check? I want to understand how bleve works as part of gitea. File size limits are available, but here I clearly fit in and it selectively searches in directories with a dot at the beginning of the name

**Comment by lunny** [comment]
I confirmed this is a bug. I will do some investigation.

**Comment by amix307** [comment]
@lunny any news?

**Comment by amix307** [comment]
On 1.24.1 same problem

**Comment by amix307** [comment]
@lunny 1.24.3 the same(

**Comment by amix307** [comment]
Resolved) Set REPO_INDEXER_EXCLUDE_VENDORED = false in app.ini under [indexer], keep REPO_INDEXER_ENABLED = true and REPO_INDEXER_TYPE = elasticsearch.
Restart Gitea, drop the old ES index (curl -XDELETE .../gitea_codes.vN) and let Gitea rebuild it; wait until the queue is empty and the alias gitea_codes → gitea_codes.vN appears. After re‑indexing, search now returns files inside .teamcity/
